### PR TITLE
feat(HDAudio): Initial implementation.

### PR DESCRIPTION
**Pull request by muscat1**

### CODE REVIEW

File: react/features/base/lib-jitsi-meet/actions.js [hunk 1]
```diff
@@ -42,11 +42,25 @@ export function initLib() {
...
 
+        if (enableHdAudio) {
+            Object.assign(config, {
+                ...config,
+                channelCount: 2,
+                disableAP: true,
+                enableNoAudioDetection: false,
+                enableNoisyMicDetection: false,
+                enableTalkWhileMuted: false,
+                opusMaxAverageBitrate: 510000,
```
Comment (muscat1) [inline react/features/base/lib-jitsi-meet/actions.js:59]: This is the maximum supported value, can be tweaked if we want.

**Comment by muscat1** [comment]
Jenkins please test this please

File: react/features/base/lib-jitsi-meet/actions.js (hunk 1 above)
Comment (saghul) [reply react/features/base/lib-jitsi-meet/actions.js:59]: IMHO that value is ridiculously high, but yeah, we can test.

Comment (saghul) [reply react/features/base/lib-jitsi-meet/actions.js:59]: We souldn't be doing this here, but in LJM.

**Review by saghul — CHANGES_REQUESTED** [review]
Let's move the handling of the new option to LJM. I see no reason why it needs to be here.

**Comment by muscat1** [comment]
Jenkins please test this please.

File: config.js [hunk 2]
```diff
@@ -125,6 +125,13 @@ var config = {
     // Enables support for opus-red (redundancy for Opus).
     // enableOpusRed: false,
 
+    // Specify audio quality stereo and opusMaxAverageBitrate values, both necessary in order to enable HD audio.
+    // Beware, by doing so, you are disabling echo cancellation, noise suppression and AGC.
+    // audioQuality: {
+    //     stereo: false,
+    //     opusMaxAverageBitrate: null // Value to fit the 6000 to 510000 range.
+    // }
```
Comment (saghul) [inline config.js:133]: Put a comma here so people don't get a syntax error when uncommenting.

File: react/features/base/config/reducer.js [hunk 3]
```diff
@@ -142,10 +142,26 @@ function _setConfig(state, { config }) {
 
     // eslint-disable-next-line no-param-reassign
     config = _translateLegacyConfig(config);
+    const hdAudioOptions = {};
+    const { audioQuality } = config;
+
+    if (audioQuality?.stereo && audioQuality?.opusMaxAverageBitrate) {
```
Comment (saghul) [inline react/features/base/config/reducer.js:148]: Let's drop the `opusMaxAverageBitrate` check.

File: react/features/base/config/reducer.js [hunk 4]
```diff
@@ -142,10 +142,26 @@ function _setConfig(state, { config }) {
...
+        const { opusMaxAverageBitrate, stereo } = audioQuality;
+
+        Object.assign(hdAudioOptions, {
+            disableAP: true,
+            enableNoAudioDetection: false,
+            enableNoisyMicDetection: false,
+            enableTalkWhileMuted: false,
+            opusMaxAverageBitrate,
```
Comment (saghul) [inline react/features/base/config/reducer.js:156]: Pass audioqueality object here.

File: react/features/base/config/configWhitelist.js [hunk 5]
```diff
@@ -15,6 +15,7 @@ export default [
     'abTesting',
     'analytics.disabled',
     'audioLevelsInterval',
+    'audioQuality',
```
Comment (saghul) [inline react/features/base/config/configWhitelist.js:18]: Remove the old stereo and opusmaxavgbitrate options.

File: react/features/base/config/reducer.js [hunk 6]
```diff
@@ -142,10 +142,26 @@ function _setConfig(state, { config }) {
 
     // eslint-disable-next-line no-param-reassign
     config = _translateLegacyConfig(config);
```
Comment (saghul) [inline react/features/base/config/reducer.js:144]: Please add some backwards compat here pl.

File: react/features/base/config/configWhitelist.js [hunk 7]
```diff
@@ -141,7 +142,6 @@ export default [
     'notifications',
     'openBridgeChannel',
     'openSharedDocumentOnJoin',
-    'opusMaxAverageBitrate',
```
Comment (saghul) [inline react/features/base/config/configWhitelist.js:144]: Sorry, my bad. Leave this in so the backwards compat code works.

File: react/features/base/config/configWhitelist.js [hunk 8]
```diff
@@ -157,7 +157,6 @@ export default [
     'startVideoMuted',
     'startWithAudioMuted',
     'startWithVideoMuted',
-    'stereo',
```
Comment (saghul) [inline react/features/base/config/configWhitelist.js:160]: Ditto.

**Review by saghul — COMMENTED** [review]
LGTM minus 2 things, my fault, sorry.

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Jenkins please test this please.

**Comment by bgrozev** [comment]
@saghul Please squash changes like this (when some commits have non-descriptive messages)

**Pull request merged**